project/glossary.md

```diff
@@ -0,0 +1,46 @@
+<!-- vim: ft=markdown colorcolumn=72
+-->
+# Glossary
+
+This glossary is a dictionary of terms defined as they are used by the
+spire/helm-charts project.  Writing the definitions of terms here helps
+to keep other documents concise and precise.  Documenting terminology
+here helps prevent misundersandings, facilitating easy onboarding of new
+team members.
+
+**Deployment**
+: A use of the Helm Charts to describe a single SPIRE cluster.
+
+**Deployment Type**
+: One of three supported deployments of the template: Standalone, 
+Primary, or Secondary.
```

Comment:
I can think of at least one more configuration that security conscious users might want... spire-server in one (security focused) k8s cluster and the spire-agent in a workload cluster. What would we call that?

Reply:
Two deployments of the charts (the charts might be identical or not, depending on how they decide to deploy)

Reply:
How would the deployment flag be set for each case? Still standalone or in some cases secondary, where agent is disabled in one release and server is disabled in the other?

I guess, is the deployment type referring to the configuration of the spire-server, or does it include agents. If it includes agents, there probably needs to be a few more permutations.